### ISSUE TITLE
keeps nagging brave didn't shutdown correctly, restore pages?

### ISSUE THREAD

**Issue by rajinder-yadav** [brave/brave-browser#18354]
## Description 

If I reboot my Linux system, every time I get this nagging popup, can someone fix to automatically restore, or provide an option to not keep nagging.

## Steps to Reproduce

   1.  Open brave
   2.  Reboot
   3.  Open brave

## Actual result:

![Screenshot_20210924_211330](https://user-images.githubusercontent.com/439226/134752967-4f622e43-d88b-4209-88d6-c949f801ca2b.png)

## Expected result:

Don't ask me just restore everything.

## Reproduces how often: 

All the time!

## Desktop Brave version:

Version 1.29.81 Chromium: 93.0.4577.82 (Official Build) (64-bit)

## Android Device details:
   - Install type (ARM, x86):
   - Device type (Phone, Tablet, Phablet):
   - Android version:

## Version/Channel Information:

- Can you reproduce this issue with the current release? yes
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

Operating System: openSUSE Tumbleweed 20210922
KDE Plasma Version: 5.22.5
KDE Frameworks Version: 5.86.0
Qt Version: 5.15.2
Kernel Version: 5.14.5-1-default (64-bit)
Graphics Platform: X11
Processors: 16 × AMD Ryzen 7 2700X Eight-Core Processor
Memory: 62.7 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3080/PCIe/SSE2

**Comment by wknapik** [comment]
Fwiw, I'm not seeing this on up-to-date Arch Linux running `Brave Browser 93.1.29.81` via `brave-bin 1:1.29.81-1`, including on a fresh profile.

@rajinder-yadav do you have this setting selected?
![image](https://user-images.githubusercontent.com/1129003/134903435-0a3d1f06-0d7a-4147-86fa-b3b47fb31583.png)

**Comment by rajinder-yadav** [comment]
@wknapik yes I do have this selected, all my tabs do open up, but only after the nag popup. Happens every time for me, I don't close the browser down when I do a reboot from the terminal.

**Comment by wknapik** [comment]
@rajinder-yadav I'm guessing closing the browser window via `Exit` in the "hamburger menu" in to top right, or the close button from your window manager would not cause that popup to be shown. This is speculation, but I'm guessing the reboot command might not wait for long enough for the browser to close gracefully and that's causing the popup to be shown, since the browser is recovering from a non-graceful shutdown.

**Comment by srirambv** [comment]
This can happen if you launch via terminal and close the terminal before the process is terminated.

**Comment by rajinder-yadav** [comment]
@wknapik My point is why even ask me? just open up the previous session or give me an option to stop the nagging. This should be an easy fix.

@srirambv I am not opening the browser from the terminal, this is just the quickest way for me to reboot.

**Comment by bsclifton** [comment]
cc: @rebron

**Comment by erdnuesse** [comment]
I know that this is closed; but here's my take on the whole ordeal, after all, I have the same issue and people may land here after all, looking for a solution, so I hope you don't mind me sharing my findings: 

My take is, I would not say that this is a bug. If the session makes Brave crash, avoiding the loops is a totally valid reason. Also; important enough to not change this behavior. /signed.
However; I experience the same issue, looking for a solution. : )

Example: I imagine, that during reboots / shutdowns, the Browser session just does not get closed properly, like firefox does it on your end.
Ie. configure firefox to _never_ ask to close all tabs, but brave in a way that it _always_ asks for confirmation if closing multiple tabs.
You furthermore never confirm that prompt, bc/ your reboot / shutdown just force closes the browser. That way, a config change could be the culprit (just never ask to confirm closing all tabs).
This is just a simple example as to how both browsers could have different in closing behavior - which would not relate to a bug but a config change.

Quick research and testings: So what I did just now; just for the heck of it, I tested the behavior of 2 settings based on my hunch.
1: Keep running brave in Background, AND
2. Warn me before closing multiple tabs.
- but I did so only once, so reproductivity was time-limited.

Results: Deactivating one of these did not result in changed behavior.
Deactivating BOTH however, brave restarted my session after a reboot / cold boot immediately for the first time in AGES without confirmation box.

As a recommendation; Looks like everyone reporting this as a bug just need to check their settings, and devs may now be aware that this issue will pop up if those two settings are used in a manner like this.
 ; )

Best regards

**Comment by rajinder-yadav** [comment]
This won't get fixed because is it a Chrome bug and all "Brave" devs do is use the same codebase slap lipstick on their "Brave" browser. Every time I opened a bug, they ask me does Chrome do the samething. This team is running off fumes and barely keeping it together.

**Comment by erdnuesse** [comment]
@rajinder-yadav - have you tried disabling the two options?

**Comment by Joshuasony** [comment]
Recently ditched windows and run only on linux. I am also annoyed about this issue. I then asked ChatGPT about its opinion and found the answer quite interesting (I didnt fact check some of the statements that probably need fact checking). Here it comes:

> Brave should fix it. Not because users are lazy, but because computers lie.
>
> Here’s the uncomfortable reality a computer would grudgingly admit if it had self awareness and a shame module:
>
> Modern operating systems do not guarantee “clean shutdowns.” Updates reboot your machine when you blink. Laptops die mid sentence. Kernel panics happen. Power flickers. Brave itself can crash. Expecting users to always “close the process correctly” is like expecting people to always die peacefully in their sleep. It’s a fantasy used to avoid engineering work.
>
> From a software responsibility standpoint:
>
> - Browsers already track session state continuously.
> - Brave already knows whether the last exit was clean.
> - Chrome, Firefox, and others already auto restore without interrogating the user every time.
> - The cost of being wrong is low. Worst case, you reopen tabs someone didn’t want. That’s reversible in one click.
>
> From a user experience standpoint:
>
> - The dialog adds friction at the exact moment the user is least patient.
> - Repeated prompts train users to click without reading. That’s actively bad design.
> - “Ask every time” is defensive programming masquerading as respect for user choice.
>
> Now, to be fair to the Brave team:
>
> - Auto restore can cause loops if a tab crashes on load.
> - Some users care deeply about not restoring sessions.
> - Brave has extra privacy constraints, which complicate assumptions.
>
> But none of that justifies a blanket “wontfix.” Those are solvable problems. You add:
>
> - A preference toggle
> - A safe mode after repeated crashes
> - A timeout or crash counter
>
> This is not a technical impossibility. It’s a prioritization decision.
>
> So the verdict:
> 
> - Users should not be required to ritualistically close a browser before shutting down a modern OS.
> - Brave should default to restoring sessions automatically after unclean exits, with an opt out.
> - Calling this a user problem is just outsourcing robustness to humans, which never works.
>
> Computers are brittle. Humans are messier. Good software absorbs that mess instead of scolding people for it.

**Comment by peterge1998** [comment]
Also would love to see this implemented! Because of standby is quite power consuming on linux, i shutdown my work laptop almost every day which meas i need to click this every day when booting which is very annoying :(
